// Copyright 2018 The StateChart Authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto2";

//import "net/proto2/util/proto/json_format.proto";

package state_chart;

message TestA {
  message TestB {
    optional int32 a_int32 = 1;
    optional int64 a_int64 = 2 [default = 10];
    optional float a_float = 3;
    optional double a_double = 4;
    optional string a_string = 5;
  }

  optional bool a_bool = 1;
  required string a_required_string = 2;
  optional TestB test_b = 3;

  extensions 15 to 20;
}

message TestAExtension {
  optional int32 ext_value = 1;

  extend TestA {
    optional TestAExtension ext = 15 [json_name = "test_a_ext"];
  }
}
